### ISSUE TITLE
Support more environment conversions

### ISSUE THREAD

**Issue by halostatue** [halostatue/enviable#5]
See [jetenv/decode.ex](https://github.com/danj3/jetenv/blob/main/lib/jetenv/decode.ex) for ideas.

Because this is meant to be used in configuration where the input is controlled, we want to support both `fetch_env_atom!` and `fetch_env_safe_atom!`.

See as well [jetenv.ex:7-34](https://github.com/danj3/jetenv/blob/0be138cc384f104a055b4af8bf7c24261bc3e8ec/lib/jetenv.ex#L7-L34)

````elixir
  @moduledoc """
  …
  # Types supported

  * S - String
  * A - Atom
  * I - Integer
  * B - boolean (true or TRUE is true all else is false)
  * F - Float
  * M - Module name
  * C - charlist
  * J - JSON encoded string
  * G - Erlang term
  * T - Base64 encoded Erlang term
  * PEM - PEM string, see note below

  Type specifiers are required suffixes, two underscores and the
  type, e.g.:

  ```
  je__my_app__some_string__S="a string value"
  ```

  ### PEM handling

  PEM strings are decoded using :public_key, however the usual
  consumers of the results (private key or cert) expect
  different things than what :public_key returns. The PEM
  handler provides values appropriate for use in `ssl_opts`
  `cacert` and `key`.
  …
  """
````

**Comment by halostatue** [comment]
It might also be useful to support a `fetch_env_log_level` that compares against `Log.levels/0` and falls back to a provided default.

Alternatively, it *might* be better to deprecate `*_boolean` and `*_integer` in favour of some sort of `*_transform` function set that incorporates the same. Requires some thought, as the function names get increasingly long.

**Comment by halostatue** [comment]
Released 1.1.0 resolves this, except `:log_level` conversion.
